python_modules/dagster/dagster/_core/definitions/repository_definition/repository_data_builder.py

```diff
@@ -215,6 +235,8 @@ def build_caching_repository_data_from_list(
                     f"Duplicate definition found for unresolved job '{definition.name}'"
                 )
             # we can only resolve these once we have all assets
+            if definition.partitions_def is not None:
+                _validate_partitions_definition(definition.partitions_def)
             unresolved_jobs[definition.name] = definition
```

Comment:
is there a way we can consolidate these three calls to `_validate_partitions_definition` into one?

Reply:
ok, I changed to collect all unique partitions definitions and then do one pass